activemodel/lib/active_model/dirty.rb

```diff
@@ -130,6 +131,22 @@ module Dirty
       attribute_method_affix prefix: "restore_", suffix: "!"
     end
 
+    def initialize_dup(other) # :nodoc:
+      super
+      @attributes = self.class._default_attributes.map do |attr|
+        attr.with_value_from_user(@attributes.fetch_value(attr.name))
+      end
+      @mutations_from_database = nil
+    end
+
+    def changes_applied # :nodoc:
+      @previously_changed = changes
+      @mutations_before_last_save = mutations_from_database
+      @attributes_changed_by_setter = ActiveSupport::HashWithIndifferentAccess.new
```

Comment:
Can we eliminate this ivar entirely now and just use the computed changes?

Reply:
It's also needed for Dirty to continue to support old ivar backed attributes.